src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryPlanManager.java

```diff
@@ -25,95 +23,140 @@
  * {@code Plan}. {@link DefaultRecoveryPlanManager} tracks currently failed (permanent) and stopped (transient) tasks,
  * generates a new {@link DefaultRecoveryBlock} for them and adds them to the recovery Plan, if not already added.
  */
-public class DefaultRecoveryPlanManager implements PlanManager {
+public class DefaultRecoveryPlanManager extends ChainedObserver implements PlanManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRecoveryPlanManager.class);
 
     protected volatile Plan plan;
     protected volatile PhaseStrategy phaseStrategy;
     protected volatile Map<UUID, PhaseStrategy> phaseStrategies;
 
     private final StateStore stateStore;
-    private final TaskFailureListener failureListener;
```

Comment:
@mgummelt:  It's used here to mark Tasks as permanently failed correct? https://github.com/mesosphere/dcos-commons/blob/0df897784b8e68e42a31c09c9af79ea25ed3fc01/src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryPlanManager.java#L275